src/data/tie_break.py

```diff
@@ -1,534 +1,1604 @@
+                raise ValueError(
+                    f'Option [{option.id}] not available '
+                    f'for tie-break [{self.name}]'
                 )
-            case PapiTieBreak.WINS:
-                return TieBreak(TieBreakType.WINS, {})
-            case PapiTieBreak.KASHDAN:
-                return TieBreak(
-                    TieBreakType.KASHDAN, {TieBreakOption.PAPI_LEGACY: True}
+            if not isinstance(option.value, option.type):
+                raise ValueError(
+                    f'Option [{option.id}={option.value}] '
+                    f'does not match expected type [{option.type}]'
                 )
-            case PapiTieBreak.KOYA:
-                return TieBreak(TieBreakType.KOYA, {})
-            case PapiTieBreak.SONNENBORN_BERGER:
-                return TieBreak(TieBreakType.SONNENBORN_BERGER, {})
-            case PapiTieBreak.PROGRESSIVE:
-                return TieBreak(TieBreakType.PROGRESSIVE_SCORES, {})
-            case _:
-                raise ValueError(f'Unknown tie break: {self}')
+            used_option_types.append(option_type)
+
+    def _get_option(
+        self, option_type: type[AbstractTieBreakOption]
+    ) -> AbstractTieBreakOption:
+        return next(
+            (
+                option for option in self.options
+                if isinstance(option, option_type)
+            ),
+            option_type(),
+        )
+
+    def _get_option_values(self) -> list:
+        """Retrieves for each of the available options
+        the corresponding value in an ordered list."""
+        return [
+            self._get_option(option_type).value
+            for option_type in self.available_options
+        ]
 
     @staticmethod
-    def _papi_buchholz_cut(tournament_rounds: int) -> int:
-        if tournament_rounds <= 7:
-            return 1
-        elif tournament_rounds <= 12:
-            return 2
-        return 3
+    def adjusted_score(
+        player: 'Player',
+        tournament: 'Tournament',
+        max_round: int | None = None,
+        adjust_fore: bool = False,
+    ) -> float:
+        """Computes the adjusted score of the player for the purposes of their opponents' tie-breaks
+        Only adjusts them in case of requested byes followed by all VUR.
+        If *adjust_fore* is True, the adjusted score for Fore Buchholz is computed:
+        games not already determined are considered a draw."""
+        if max_round is None:
+            max_round = max(player.pairings)
+        if tournament.pairing == TournamentPairing.BERGER:
+            return player.points_after(max_round)
+        score = 0
+        for round_index, pairing in player.pairings.items():
+            if round_index > max_round:
+                continue
+            if adjust_fore and round_index == max_round:
+                if pairing.result in (
+                    Result.FULL_POINT_BYE,
+                    Result.PAIRING_ALLOCATED_BYE,
+                ):
```

Comment:
Games not determined over the board use their result's points
```suggestion
                if pairing.result in (
                    Result.FULL_POINT_BYE,
                    Result.PAIRING_ALLOCATED_BYE,
                    Result.HALF_POINT_BYE,
                    Result.ZERO_POINT_BYE,
                ):
```

Reply:
I commited this suggestion, but it fails the fore_buchholz tests. I'm reverting it for now, letting you handle this if that's ok